### DISCUSSION TITLE
connect econnrefused "127.0.0.1:80"

### DISCUSSION THREAD

**Discussion by alimttth** [vercel/next.js#62784] [Help] [answered]
### Summary

`export async function getServerSideProps(context) {
    const searchParams = {
        startDate: moment().subtract(7, 'd').toDate(),
        endDate: moment().toDate(),
    }
    const res = await axios.get(`/api/trends/date?.start=${searchParams.startDate.toISOString()}&end=${searchParams.endDate.toISOString()}&geo=${searchParams.geo}`)
    return {
        props: {data: res}
    }
}`

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I think there's two errors here.

1. Reaching to `/api/trends/date` which is a relative path, and then you must ask yourself, relative to what, a plain Node.js script has no concept of URL. Typically people provide globals, envs, and such to over come this issue, or let the http/https node modules resolve that, but the real fix would be to make this request absolute. 

However, that'd be inefficient and unnecessary, because you are already on the server! Also code used only within GetServerSideProps, is dropped from the client bundle, so you can connect with your 3rd party API directly, within GetServerSideProps:

- https://next-code-elimination.vercel.app/

And, said by docs themselves, https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props#behavior:

> You do not have to call a Next.js [API Route](https://nextjs.org/docs/pages/building-your-application/routing/api-routes) to fetch data when using getServerSideProps since the function runs on the server. Instead, you can call a CMS, database, or other third-party APIs directly from inside getServerSideProps.

2. The lesser problem is that, the axios request is probably resolving to a an IPV6 address.

So the fix, is to follow this guide:

- https://nextjs-faq.com/fetch-api-in-getserversideprops

It explains the problem, and the solution. Also, the way you have it now, anyone could try and reach to `/api/trends/date`, and that may not be what you want at all.

**Reply by alimttth** [reply]
Thank you, problem solved
